### PR TITLE
Adds gas rounds to MLRS 

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request

https://github.com/tgstation/TerraGov-Marine-Corps/assets/46353991/c978e981-d88a-4850-a560-4a5b122556cb

Adds Gas rounds to the MLRS which add a smoke cloud effect that drains plasma and deals damage to anything inside of it, pretty much like WP without the fire effect.
## Why It's Good For The Game
Adds an interesting round type to MLRS to let it shine more in assault situations and also gives it something unique compared to other artillery pieces.
## Changelog
:cl:
add: MLRS 'X-50' Gas rockets.
/:cl:

### CODE REVIEW

**Comment by maikilangiolo** [comment]
> it shine more in assault situations

Not that it's bad, but I fail to see how a lingering cloud of damaging smoke, which hinders marines, is better at assaults than an instant missile barrage.

**Comment by Naaanii** [comment]
Tfoot would be based.

File: code/game/objects/effects/effect_system/smoke.dm [hunk 1]
```diff
@@ -257,6 +257,12 @@
 	color = "#DBCBB9"
 	smoke_traits = SMOKE_GASP|SMOKE_BLISTERING|SMOKE_OXYLOSS|SMOKE_PLASMALOSS|SMOKE_FOUL
 
+/obj/effect/particle_effect/smoke/phosphorus/mustard
+	opacity = TRUE
+	color = "#FFDB58"
```
Comment (KirovClassBattlecruiser) [inline code/game/objects/effects/effect_system/smoke.dm:262]: Why can't you have it be purple?

File: code/game/objects/machinery/mortar.dm [hunk 2]
```diff
@@ -675,6 +676,12 @@
 	icon_state = "mlrs_rocket"
 	ammo_type = /datum/ammo/mortar/rocket/mlrs
 
+/obj/item/mortal_shell/rocket/mlrs/gas
+	name = "\improper 60mm 'X-50' rocket"
+	desc = "A 60mm rocket loaded with deadly X-50 gas, meant to be used in saturation fire with high scatter."
```
Comment (KirovClassBattlecruiser) [inline code/game/objects/machinery/mortar.dm:681]: Description would not really help the average unga understand what it's all about, i propose:
desc = "A 60mm rocket for the MLRS loaded with deadly X-50 gas that drains plasma and deals damage to any poor soul standing inside it, meant to be used [How you decide it should be, either shock or area suppresion] with high scatter."

File: code/game/objects/effects/effect_system/smoke.dm [hunk 3]
```diff
@@ -257,6 +257,12 @@
...
+	lifetime = 40
```
Comment (KirovClassBattlecruiser) [inline code/game/objects/effects/effect_system/smoke.dm:263]: A rather indecise lifetime, not a good fit for a shock role or an area suppresion role, 20 or 60 would be better picks depending on what you wan to do with it

**Review by KirovClassBattlecruiser — CHANGES_REQUESTED** [review]
Well i'm happy that at least the ideasguy coded his idea in this time except i do not think it would be wise to implement is as-is, see code review

File: code/game/objects/effects/effect_system/smoke.dm (hunk 1 above)
Comment (Pariah919) [reply code/game/objects/effects/effect_system/smoke.dm:262]: Purple would overlap with tanglefoot no?

Comment (TiviPlus) [inline code/game/objects/effects/effect_system/smoke.dm:262]: Use a color define

**Review by TiviPlus — COMMENTED** [review]
rest lgtm

File: code/game/objects/effects/effect_system/smoke.dm (hunk 1 above)
Comment (KirovClassBattlecruiser) [reply code/game/objects/effects/effect_system/smoke.dm:262]: how did i miss that in my review

Comment (KirovClassBattlecruiser) [reply code/game/objects/effects/effect_system/smoke.dm:262]: i will commit sepuku

**Review by ivanmixo — APPROVED** [review]

**Pull request merged**